projects/igniteui-angular/src/lib/toast/toast.component.ts

```diff
@@ -331,31 +375,7 @@ export class IgxToastComponent implements IToggleView, OnInit, OnDestroy {
      * @memberof IgxToastComponent
      */
     public toggle() {
-        this.isVisible ? this.close() : this.open();
-    }
-
-    /**
-     * @hidden
-     */
-    @HostBinding('class.igx-toast--top')
-    public get cssClassTop() {
-       return this.position === IgxToastPosition.Top;
-    }
-
-    /**
-     * @hidden
-     */
-    @HostBinding('class.igx-toast--middle')
-    public get cssClassMiddle() {
-       return this.position === IgxToastPosition.Middle;
-    }
-
-    /**
-     * @hidden
-     */
-    @HostBinding('class.igx-toast--bottom')
-    public get cssClassBottom() {
-       return this.position === IgxToastPosition.Bottom;
+        this._isVisible ? this.close() : this.open();
```

Comment:
this will not work because of the animation. You can try to create a button and call `toggle` method by quickly clicking on the button several times. If you observe the console you will see a lot of warnings. All these is solved in `IgxToggleDirective` and you can use it for the toast.

Reply:
Should be handled better now, as it uses the `collapsed` property of the toggle directive to decide whether to call the `open` or `closed` methods.